concepts/functions/about.md

```diff
@@ -1,3 +1,176 @@
 # About
 
-TODO
+Functions are bodies of code containing one or more [statements or expressions][statements or expressions]. A function may optionally
+[return an expression][return value]. (Recall that a statement is an instruction that does not return a value. An expression evaluates to
+a resulting value.)
+
+Rust style is to use [snake case][snake case] for a function name, which is prefaced by the `fn` keyword for a function definition. For example
+
+```rust
+fn do_nothing() {}
+
+```
+
+The empty parentheses are a placeholder for zero or more parameters. A parameter is the definition of a value to be passed into the function.
+Even if there are no parameters the parentheses are still required in the function definition. The actual value passed to a parameter is called
+an argument.
+
+The combination of parameters and return value is known as the function's signature. A function's signature requires that each parameter must
+have its type annotated.
+
+The curly braces enclose the body of the function. Inside the braces are where the function's statements and expressions are placed. Even if the
+function contains no statements or expressions the curly braces are still required for the function definition. Of course there is no point to
+having an empty function.
```

Comment:
> Of course there is no point to having an empty function.

That's not entirely true. Just about an hour ago, I was working with some code which (paraphrased) worked out to

```rust
impl VeryComplicatedStruct {
  fn create_with_mutate_intermediate_value(mutate_intermediate: impl FnOnce(&mut IntermediateStruct)) -> Self;
}
```

Given that signature, if you want to create a `VeryComplicatedStruct` _without_ actually changing the intermediate value, you need to pass an empty function: `VeryComplicatedStruct::create_with_mutate_intermediate_value(|_intermediate| {})`.

It's not worth adding that entire story to this document, but we should be careful about assumptions about what is useful.

```suggestion
function contains no statements or expressions the curly braces are still required for the function definition. 
```

Reply:
You know, when I typed that I suspected there was some corner case that would prove that wrong, so your observation is not  unexpected.